### ISSUE TITLE
Change the order of the 2D/3D and Script buttons to make more sense

### ISSUE THREAD

**Issue by mieldepoche** [godotengine/godot-proposals#8212]
### Describe the project you are working on

stuff

### Describe the problem or limitation you are having in your project

current order:
 - 2D (position 1, Ctrl + F1)
 - 3D (position 2, Ctrl + F2)
 - Script (position 3, Ctrl + F3)

proposed order:
 - Script (position 1, Ctrl + F1)
 - **2**D (position **2**, Ctrl + F**2**)
 - **3**D (position **3**, Ctrl + F**3**)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

- Since the Script option is used by everyone (-C# only devs), it make sense to me that it would be in front of 2D and 3D which are not always used by everyone.
- I always used to mix the default shortcuts up: they are in the order of the buttons, but don't make sense to me.
With this change, the shorcuts are not only tied to the button position, but it also have a deeper link: 2D -> F2, 3D -> F3, Script -> F1 (not spatial in any way if you will, more primitive? point is, it doesn't not make sense)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

https://github.com/godotengine/godot-proposals/assets/71937306/4a3cfc86-b4f0-47c3-bbd7-e8426ad5d42c

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It is used often and can be worked around with a tiny addon (see below).
However seeing this is a tiny ui change that impacts my experience a lot, I want to know what others think about it.

### Is there a reason why this should be core and not an add-on in the asset library?

```gdscript
@tool
extends EditorPlugin


var button_container: Control


func _enter_tree() -> void:
    var title_bar := get_tree().root.find_child("*EditorTitleBar*", true, false)
    button_container = title_bar.get_child(2)
    button_container.move_child(button_container.get_child(2), 0)


func _exit_tree() -> void:
    button_container.move_child(button_container.get_child(0), 2)
```

**Comment by Calinou** [comment]
I share similar concerns as https://github.com/godotengine/godot-proposals/issues/7700#issuecomment-1722520863 about this proposal, since this would break a lot of muscle memory.

**Comment by Mickeon** [comment]
If I may add, just in general, these shortcuts are much more tedious to perform than clicking on their respective buttons, as the CTRL and Function keys are too far from eachother. It's worth considering something else entirely.

**Comment by Calinou** [comment]
> If I may add, just in general, these shortcuts are much more tedious to perform than clicking on their respective buttons, as the CTRL and Function keys are too far from each other. It's worth considering something else entirely.

I agree with that, but I'm not sure what shortcuts we could use that don't conflict with other OS shortcuts on Windows/macOS/Linux.

**Comment by Mickeon** [comment]
I think the Function keys are somewhat off-limits. Although, if I may say, I did map these shortcuts myself to:
- **F2** for the **2D editor**;
- **F3** for the **3D editor**;
- **F4** for the **Script Editor**;
- **Nothing** for the **Asset Library**. This is not something you really need quickly anyhow.

I feel like it makes a lot of sense. Although, muscle memory dictates I still click on the buttons.

**ALT** + **1,2,3** and **4** are fine candidates. I recall that before zooming with number keys in the 2D editor was introduced, these number keys were the shortcuts.

###### ~~We could use ALT + F1, F2, and F3. Then leave F4 as the Kinder surprise for anyone trying to access the Asset Library that way.~~

**Comment by mieldepoche** [comment]
reordering those buttons is possible in latest builds

**Comment by Calinou** [comment]
> reordering those buttons is possible in latest builds

For future reference, this was implemented by https://github.com/godotengine/godot/pull/113051. The default button order wasn't changed though.
